### PR TITLE
feat(version): add version and compile options

### PR DESCRIPTION
**Pull request by rphillips**
screen shot 2015-02-06 at 9 59 40 am

### CODE REVIEW

**Comment by rphillips** [comment]
Also fixes a few lint errors.

File: src/luvi.c [hunk 1]
```diff
@@ -18,10 +18,28 @@
...
+  lua_newtable(L);
+#ifdef WITH_OPENSSL
+  lua_pushnumber(L, i++);
+  lua_pushstring(L, "ssl");
+  lua_settable(L, -3);
+#endif
+#ifdef WITH_ZLIB
+  lua_pushnumber(L, i++);
+  lua_pushstring(L, "zlib");
+  lua_settable(L, -3);
```
Comment (creationix) [inline src/luvi.c]: Not that this code is performance sensitive, but in this case it's better to use http://www.lua.org/manual/5.1/manual.html#lua_rawseti

Then you don't need to push wrapped numbers on the stack and skips a lot of steps internally.

File: src/lua/init.lua [hunk 2]
```diff
@@ -407,7 +420,13 @@ return function(args)
     if not target or #target == 0 then return commonBundle(bundle) end
     return buildBundle(target, bundle)
   end
-  local usage = [[Luvi Usage Instructions:
+  local prefix = [[$(LUVI) $(LUVI_VERSION)]]
+  prefix = string.gsub(prefix, "%$%(LUVI%)", args[0])
+  prefix = string.gsub(prefix, "%$%(LUVI_VERSION%)", luvi.version)
```
Comment (creationix) [inline src/lua/init.lua]: Lua has string.format that works a lot like sprintf.  You should probably use that instead of gsubs.

Comment (rphillips) [inline src/lua/init.lua]: fixed

**Comment by creationix** [comment]
+1

**Pull request merged**